Runtime/Scripts/VehicleComponents/Sensors/SSS.cs

```diff
@@ -10,20 +10,29 @@ public class SideScanSonar: Sensor
     {
         Sonar sonarPort;
         Sonar sonarStrb;
+        private ushort magic_number = 20860;
+        private Color rayColor;
 
         [Header("SideScanSonar")]
         public int numBucketsPerSide = 1000;
         public byte[] portBuckets;
         public byte[] strbBuckets;
+        public byte[] portBucketsAngleHigh;
+        public byte[] portBucketsAngleLow;
+        public byte[] strbBucketsAngleHigh;
+        public byte[] strbBucketsAngleLow;
+        public bool isISSS = false;
+        public bool drawRays = false;
+
         [Header("Beam Profile")]
-        public float beamOrientationAngleDeg = 45.0f;
+        public float beamOrientationAngleDeg = 55.0f;
         public float fullBeamAngleDeg = 90.0f;
         public float fwhmBeamAngleDeg = 60.0f;
         public bool gaussianProfile = false;
         public int maxRange = 100;
-        public int totalBeamCount = 256;
+        public int totalBeamCount = 4000;
```

Comment:
4k beams by default is _way too much_. This will freeze many systems :)

Reply:
Emmm, how about 4K beams but instead of 20 hz, we have 1 hz?

Reply:
that.. is a question of how unity handles parallel jobs. I am not so certain about balancing over frames. I think as it is, it will cast all 4k beams asap leading to a spike and then wait around for the rest of the 1 second. So there'll be spikes of low frames every second _i think_. @martkartasev what think you?

Reply:
I would emphasize that we are talking about **_default_** values. There is no need for the default value to increase as it wont be relevant for most users. That doesn't mean we cannot provide for the possibility of a higher refresh rate sonar - It can currently be configured from the component.

Additionally, it seems to me that it might make sense to create the ISSS separately from the SSS? 
Or is this a straight up upgrade in the sense that the current SSS is not really representative of any real device?

4k beams runs quite OK on my machine, for the record. If performance is a concern and If the number of beams is more important than the refresh rate, there are some tricks I can apply to make this work even better albeit at a small cost of accuracy.

Essentially, we could schedule the beams over multiple frames. A 1hz rate means we have approximately 50 frames to compute - It wont take more than 4-5 on most machines. I can make this a toggleable option. 
I remember experimenting with upwards of 20k beams and was able to finish processing in a few frames normally. 
The downside is that all the beams wont hit at exactly the same moment of time in the simulation. However they will hit the same position on the ground, regardless of when exactly they are read back into the bucket - assuming there is no moving object. (@KKalem the raycast command is given with a origin and direction - those parameters wont change even if the returns happen at later frames).

We can discuss this further during today's meeting.

Reply:
Okay, I see what you're saying. Then I won't change this default value. 

As for the separation from SSS, here's my two cents: SAM is getting a new ISSS from DeepVision and it is basically a straight up upgrade. And the new data we have is the elevation angle for each sidescan bin. I think the `public bool isISSS = false;` is the most sensible solution to me.

And about the frame rate - I personally think the number of beams is more important than frame rate. Since the AUV is generally moving not very fast, high frame rate gains quite little.